deepcell/datasets/builders.py

```diff
@@ -0,0 +1,209 @@
+
+    def _pad_array(self,
+                   arr,
+                   node_axes=[1],
+                   time_axes=[2],
+                   temporal_adj=False,
+                   pad_value=0):
+
+        n_axes = len(arr.shape)
+        pads = []
+
+        max_nodes = self.max_nodes
+        max_frames = self.max_frames
+
+        if temporal_adj:
+            max_frames -= 1
+
+        for ax in range(n_axes):
+            if ax in node_axes:
+                pads.append((0, max_nodes - arr.shape[ax]))
+            elif ax in time_axes:
+                pads.append((0, max_frames - arr.shape[ax]))
+            else:
+                pads.append((0, 0))
+        pads = tuple(pads)
+        values = (pad_value, pad_value)
+        arr = np.pad(arr,
+                     pads,
+                     mode='constant',
+                     constant_values=values)
+        return arr
+
+    def _load_tracks(self):
+
+        appearances = []
+        centroids = []
+        morphologies = []
+        adj_matrices = []
+        norm_adj_matrices = []
+        temporal_adj_matrices = []
+
+        for track in self.track_list:
+            app = self._pad_array(track.appearances)
+            cent = self._pad_array(track.centroids)
+            morph = self._pad_array(track.morphologies)
+            adj = self._pad_array(track.adj_matrix,
+                                  node_axes=[1, 2],
+                                  time_axes=[3])
+            norm_adj = self._pad_array(track.norm_adj_matrix,
+                                       node_axes=[1, 2],
+                                       time_axes=[3])
+            temporal_adj = self._pad_array(track.temporal_adj_matrix,
+                                           temporal_adj=True,
+                                           node_axes=[1, 2],
+                                           time_axes=[3])
+
+            appearances.append(app)
+            centroids.append(cent)
+            morphologies.append(morph)
+            adj_matrices.append(adj)
+            norm_adj_matrices.append(norm_adj)
+            temporal_adj_matrices.append(temporal_adj)
+
+        self.appearances = np.concatenate(appearances, axis=0)
+        self.centroids = np.concatenate(centroids, axis=0)
+        self.morphologies = np.concatenate(morphologies, axis=0)
+        self.adj_matrices = np.concatenate(adj_matrices, axis=0)
+        self.norm_adj_matrices = np.concatenate(norm_adj_matrices, axis=0)
+        self.temporal_adj_matrices = np.concatenate(temporal_adj_matrices, axis=0)
+
+    def _sample_time(self, *args):
+        X_dict = args[0]
+        y_dict = args[1]
+
+        app = X_dict['appearances']
+        max_time = tf.shape(app)[1] - self.track_length
+        t_start = tf.random.uniform([1],
+                                    minval=0,
+                                    maxval=max_time,
+                                    dtype=tf.int32)[0]
+
+        for key in X_dict.keys():
```

Comment:
dictionaries iterate over the keys by default, so there is no reason to call `.keys()` here, which creates a new list.